### DISCUSSION TITLE
Feature suggestion: Cache the BibTeX file read

### DISCUSSION THREAD

**Discussion by wenjie2wang** [emacs-citar/citar#68] [General]
Thanks for this nice package!  I gave it a try and it worked smoothly for me.

I have a relatively large BibTeX file with over 1,000 entries (and 10,000 lines) and I have been using `ivy-bibtex`.  The `bibtex-ctions` takes a noticeable time to read/load the BibTeX file every time, while `ivy-bibtex` takes a noticeable time only the first time (assuming that the BibTeX file has not been changed).  I guess it is because `ivy-bibtex` somehow caches the BibTeX file that has been read the first time.  Thus, IMHO, it would be great if `bibtex-actions` contains a similar caching mechanism.

**Comment by bdarcus** [comment]
Yeah, I haven't tested with large files, but that makes sense.

I created this issue, because I agree we need to figure it out:

https://github.com/bdarcus/bibtex-actions/issues/69

**Reply by bdarcus** [reply]
Update: it's complicated.

See this post for an update: https://github.com/bdarcus/bibtex-actions/issues/69#issuecomment-817232480

**Reply by bdarcus** [reply]
Caching now added: #71, along with additional performance improvements with #79.

Turns out it wasn't that complicated :-)
